Doc/library/functions.rst

```diff
@@ -1117,9 +1117,10 @@ are always available.  They are listed here in alphabetical order.
 
 .. class:: object()
 
-   Return a new featureless object.  :class:`object` is a base for all classes.
-   It has methods that are common to all instances of Python classes.  This
-   function does not accept any arguments.
+   This is the ultimate base class of all other classes. It has methods
```

Comment:
While I get what you mean with "[object] is the ultimate base class of all other classes.", and it's probably good enough for this description, is that _actually true_?
From this little experiment in the REPL:
```python
Python 3.8.10 (default, Mar 13 2023, 10:26:41)
[GCC 9.4.0] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> type(object)
<class 'type'>
>>> isinstance(type, object)
True
>>> isinstance(object, type)
True
>>>
```
it seems to me that both `object` and `type` are equally ultimate base classes.

Reply:
You are checking the instance–class relationship here, not class–base class!

Reply:
Fair enough. I also tested with `issubclass` which confirms that behaviour.

But how can `object` be a supertype of `type` and an instance of `type` at the same time?
I'm no polymorphism expert, so there's probably a gap in my understand, but ISTM that the `__bases__` attribute is lying a bit.

Reply:
Oh it’s a special case handled by cheating!

Everything is an instance of object, so type is an instance of object.
type is the class of classes, so object is an instance of types.

These properties are the goal, so the implementations cheat to make it work.

Reply:
Sounds like the real question is which came first, the `object` or the `type`? :laughing:

Reply:
The Dutch?